### PR TITLE
Rdsimaps3333 fix duplicate autocomplete

### PR DESCRIPTION
**Pull request by acarrera42**
This PR also has code changes from https://github.com/bloomberg/cartodb/pull/251 since the unit tests came from that change.  Therefore once it is merged this PR may appear shorter.  

@gfiorav for review as well.

### CODE REVIEW

File: spec/connectors/importer_spec.rb [hunk 1]
```diff
@@ -8,6 +8,10 @@
 
   before(:all) do
     @user = create_user(quota_in_bytes: 1000.megabyte, table_quota: 400, max_layers: 4)
+    @common_data_user = Carto::User.where(username: Cartodb.config[:common_data]['username']).first
+    unless @common_data_user
+      @common_data_user = FactoryGirl.create(:carto_user, { username: Cartodb.config[:common_data]['username'] } )
+    end
```
Comment (gfiorav) [inline spec/connectors/importer_spec.rb]: Always do the FactoryGirl. No reason no to, and that way you don make assumptions on app_config (that has already caused some problems!) :)

File: app/connectors/importer.rb [hunk 2]
```diff
@@ -77,7 +77,9 @@ def register(result)
           name = result.name
         # HACK - Samples 2.0 Save As - The actual runner of some tables are hidden because they arecreated under the scenes
         #  Longer term a new runner should be created and additional info needs to be passed up
-        elsif result.schema != ORIGIN_SCHEMA && File.extname(@runner.downloader.source_file.filename) == '.carto'
+        elsif result.schema != ORIGIN_SCHEMA &&
+            ((@runner.downloader.source_file.filename && File.extname(@runner.downloader.source_file.filename) == '.carto') ||
+             File.extname(@runner.downloader.source_file.path) == '.carto')
```
Comment (gfiorav) [inline app/connectors/importer.rb]: To remove these changes, maybe you can base this branch off the other one you're going to merge.

**Review by gfiorav — COMMENTED** [review]
Left some comments!

File: app/connectors/importer.rb [hunk 3]
```diff
@@ -77,7 +77,9 @@ def register(result)
           name = result.name
         # HACK - Samples 2.0 Save As - The actual runner of some tables are hidden because they arecreated under the scenes
         #  Longer term a new runner should be created and additional info needs to be passed up
-        elsif result.schema != ORIGIN_SCHEMA && File.extname(@runner.downloader.source_file.filename) == '.carto'
+        elsif result.schema != ORIGIN_SCHEMA &&
+            ((@runner.downloader.source_file.filename && File.extname(@runner.downloader.source_file.filename) == '.carto') ||
```
Comment (mbektash) [inline app/connectors/importer.rb]: it would be better to create a function like `def source_extension` and then source_extension == '.carto'

**Comment by gfiorav** [comment]
Could you update this once the other branch is merged? that way it's easier to focus on the changes! Thanks :)

**Comment by acarrera42** [comment]
The review has been updated after the merge.  It should be easier to see the exact change for this fix now.

**Review by gfiorav — APPROVED** [review]
Super cool! Thanks! :+1:

**Pull request merged**